### PR TITLE
Add helpful caveats to the zsh formula

### PR DESCRIPTION
**Pull request by mnin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/zsh.rb [hunk 1]
```diff
@@ -62,6 +62,17 @@ def install
     end
   end
 
+  def caveats; <<~EOS
+    You will need to add:
+      #{HOMEBREW_PREFIX}/bin/zsh
+    to /etc/shells.
+
+    Then run:
+      chsh -s #{HOMEBREW_PREFIX}/bin/zsh
```
Comment (SMillerDev) [inline Formula/zsh.rb]:
```suggestion
      chsh -s #{bin}/zsh
```

File: Formula/zsh.rb [hunk 2]
```diff
@@ -62,6 +62,17 @@ def install
     end
   end
 
+  def caveats; <<~EOS
+    You will need to add:
+      #{HOMEBREW_PREFIX}/bin/zsh
```
Comment (SMillerDev) [inline Formula/zsh.rb]: Afaik this isn't really needed.

File: Formula/zsh.rb (hunk 1 above)
Comment (mnin) [reply Formula/zsh.rb]: Thanks for the suggestion.

**Comment by lembacon** [comment]
This was considered as non-Homebrew-specific, and had been removed in https://github.com/Homebrew/homebrew-core/commit/bcc36757114b211e8e69b2dbb60975a0cdb0492f.

**Comment by mnin** [comment]
@lembacon in my honest opinion, this is very helpful to use zsh unix conform. With the fish shell there is exactly this hint: https://github.com/Homebrew/homebrew-core/blob/master/Formula/fish.rb#L43

**Comment by MikeMcQuaid** [comment]
Caveats are warnings not documentation, sorry: https://docs.brew.sh/Formula-Cookbook#caveats

**Comment by mnin** [comment]
@MikeMcQuaid interesting, I understand this as a warning, so that the user knows how to change the shell and remain Unix compliant. But with this interpretation, we should also delete the caveats on the fish shell.

**Comment by mnin** [comment]
Additionally, on the mysql formula, the caveats looks also as documentation. https://github.com/Homebrew/homebrew-core/blob/master/Formula/mysql.rb#L108 Everyone should know how to run the mysql console.

**Comment by SMillerDev** [comment]
Yup, some have slipped through. Could you make a pull request to remove those?

**Comment by mnin** [comment]
@SMillerDev @MikeMcQuaid wouldn't we be throwing away meaningful help for the users then?

**Comment by MikeMcQuaid** [comment]
That's not what caveats are for.

**Comment by mnin** [comment]
@MikeMcQuaid that means, you agree to remove the caveats e.g. on fish and mysql formula, correct?

**Comment by jonchang** [comment]
> @MikeMcQuaid that means, you agree to remove the caveats e.g. on fish and mysql formula, correct?

> Yup, some have slipped through. Could you make a pull request to remove those?

**Comment by mnin** [comment]
I think we should introduce a new section for these informations called "hint". That's where this information should go.

I would not like to remove them without replacement, because it always helped quickly.

On top of that I expect helpless and dissatisfied users, when we simply remove all of the caveats.

**Comment by SMillerDev** [comment]
I don't think writing manuals is the task of the package management tool. @jonchang made a PR already to fix the abuse of the `caveat` block by other shells.

**Comment by mnin** [comment]
@SMillerDev good to know, but @jonchang forgot to remove the infos from mysql as well...

**Comment by SMillerDev** [comment]
That's not a shell. I can make a PR for that later though. I'm not sure if those caveats represent the default install style though. I know they're similar to ubuntu, but it might still be non-default

**Comment by MikeMcQuaid** [comment]
> I don't think writing manuals is the task of the package management tool.

This. The `caveats` are there to document Homebrew specific limitations (check the definition of word "caveats").

**Pull request closed without merging**